docs/development/quickstart.rst

```diff
@@ -0,0 +1,365 @@
+- **Test changes**: For code changes, ensure that they work as expected following the
+  process outlined in the :ref:`testing-guidelines` section.
+- **Build documentation**: If you are updating the documentation, you will want to
+  :ref:`build the documentation <builddocs>` to ensure that it looks good.
+- **Add a changelog entry**: For most code changes you will need to
+  :ref:`add-changelog`.
+
+.. tip:: For more information and examples see :ref:`edit-flow` section.
+
+You can see all the changes you've currently made by running:
+
+.. code-block:: shell
+
+    git status
+
+You can then commit your all your changes to your local repository with an explanatory
+commit message:
+
+.. code-block:: shell
+
+    git add files-that-you-changed ...
+    git commit -m "your commit message goes here"
+
+You should make frequent commits and always include a commit message. Each commit
+should represent one logical set of changes.
+
+.. Important:: Never merge changes from ``upstream/main`` into your feature branch. If
+   changes in ``main`` require changes to our code you must :ref:`rebase`.
+
+.. _contributing.push-code:
+
+Pushing your changes
+--------------------
+
+When you want your changes to appear publicly on your GitHub page, push your
+forked feature branch's commits::
+
+    git push origin subpackage-new-feature
+
+Here ``origin`` is the default name given to your remote repository on GitHub.
+
+
+
+You can see the remote repositories
+
+.. code-block:: shell
+
+    git remote -v
+
+If you added the upstream repository as described above you will see something
+like
+
+.. code-block:: shell
+
+    origin  git@github.com:yourname/astropy.git (fetch)
+    origin  git@github.com:yourname/astropy.git (push)
+    upstream        https://github.com/astropy/astropy.git (fetch)
+    upstream        https://github.com/astropy/astropy.git (push)
+
+Now your code is on GitHub, but it is not visible to the Astropy maintainers. For that
+to happen, a pull request needs to be submitted on GitHub.
+
+The first time you push to a new branch on GitHub, you will see a message like below
+with a useful link to create a pull request:
+
+  remote: Create a pull request for 'subpackage-new-feature' on GitHub by visiting:
+  remote:      https://github.com/yourname/astropy/pull/new/subpackage-new-feature
+
+
+.. _quickstart-pull-request:
+
+Making a pull request
+---------------------
+
+If everything looks good, you are ready to make a pull request (PR). A PR is how
+code from your local repository becomes available to the GitHub community to review and
+merged into project to appear the in the next release.
+
+Most of the time you can just follow the link that ``git`` provide when you pushed
+your branch and create the PR. If you don't have that link (and for a few more details), you can follow the :ref:`pull-request` instructions.
+
+Follow the instructions in the PR template and fill it out as completely as possible.
```

Comment:
Should this say something about labels? Or can normal users not set those?

Reply:
Every user can **see** labels, but only maintainers can **set** them.